experimental/entity/entity.proto

```diff
@@ -0,0 +1,219 @@
+    // The version string is not sortable with other versions
+    string version = 4; 
+
+    // Content checksum
+    string etag = 5; 
+
+    // Size in bytes
+    int64 size = 6;
+
+    // list of operations the calling user can perform
+    repeated string allowedOperations = 7; 
+
+    // avaliable PRs for this path
+    repeated EntityPR prs = 8; 
+}
+
+message EntityPR {
+    // PR Title
+    string name = 1;
+
+    // passed to get and it will load the content
+    string version = 2; 
+
+    // Unix millis for last update time
+    int64 updatedAt = 3;
+
+    // External URL
+    string url = 4;
+}
+
+message FolderListing {
+    string path = 1; 
+
+    // Child entity information
+    repeated EntityMessage items = 2;
+
+    // More items exist
+    string nextPageToken = 3;
+}
+
+message EntityHistory {
+    // the version that will load the entity object
+    string version = 1; 
+    // Unix millis when the event happened
+    int64 when = 2; 
+    // Identifier for the source 
+    string who = 3; 
+    // optional path to source location
+    string comment = 4; 
+}
+
+message EntityHistoryResponse {
+    string path = 1;
+    
+    repeated EntityHistory event = 2;
+
+    // optional path to source location
+    string nextPageToken = 3; 
+}
+
+message KindSummary {
+    string ID = 1;
+    string description = 2;
+    string category = 3;
+
+    // Detect the kind from a path suffix
+    string pathSuffix = 4;
+
+    // The go package type for this kind
+    string goType = 5;
+
+    // For kinds with secure keys -- the keys will be strpped unless user has editor access
+    bool hasSecureKeys = 6;
+
+	// The entity store does not extend the base EntityEnvelope -- this is typical for
+	// non-object-model formats like images (png, svg, etc)
+    bool isRaw = 7;
+
+    // For raw content types, this is set as an HTTP header
+    string contentType = 8;
+}
+
+//---------------------------------------------------------
+// Resource service enables HTTP-style requests over gRPC.
+//---------------------------------------------------------
+
+message GetEntityRequest {
+    // Path encodes the UID + kind.  The schema registry will know how to map suffix to registered kinds
+    string path = 1; 
+
+    // Version specifies which version of the 
+    string version = 2;  
+
+    // Include storage metadata in the response
+    bool withStorageMeta = 3;
+
+    // Include the entity payload in response
+    bool withPayload = 4;
+
+    // Augment results with access restrictions
+    bool withACL = 5;
+
+    // Include a list of alternative versions (support depends on storage engine) 
+    bool withPRs = 6;
+}
+
+message ListFolderRequest {
+    string path = 1; 
+
+    string filterKind = 2;
+
+    // Include storage metadata for each item
+    bool withStorageMeta = 3;
+
+    // Include the entity payload for each item
+    bool withPayload = 4;
```

Comment:
@torkelo -- you are right, this could be "slow" as it look up each object.  Perhaps we should exclude it?  We need it for things like indexing from scratch -- where we need an iterator over document bodies.

This should not be necessary for a list view